### PR TITLE
Integrate @ShouldPin/@ShouldNotPin into QuarkusTests

### PR DESCRIPTION
**Pull request by cescoffier**
This PR brings the functionality from loom-unit (partially, but the important ones are there) into Quarkus.
Thus, @ShouldPin and @ShouldNotPin can be used on a @QuarkusTest directly as soon as the quarkus-junit5-virtual-threads dependency is in the project.

loom-unit original repo: https://github.com/cescoffier/loom-unit

### CODE REVIEW

**Review by geoand — APPROVED** [review]
Great stuff!

File: test-framework/junit5-virtual-threads/src/main/java/io/quarkus/test/junit/virtual/internal/InternalEvents.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+public interface InternalEvents {
+
+    String INITIALIZATION_EVENT_NAME = "io.quarkus.test.junit.virtual.internal.InternalEvents.InitializationEvent";
+    String SHUTDOWN_EVENT_NAME = "io.quarkus.test.junit.virtual.internal.InternalEvents.ShutdownEvent";
+
+    String CAPTURING_STARTED_EVENT_NAME = "io.quarkus.test.junit.virtual.internal.InternalEvents.CapturingStartedEvent";
+    String CAPTURING_STOPPED_EVENT_NAME = "io.quarkus.test.junit.virtual.internal.InternalEvents.CapturingStoppedEvent";
+
+    @Name(INITIALIZATION_EVENT_NAME)
+    @Category("loom-unit")
```
Comment (ozangunalp) [inline test-framework/junit5-virtual-threads/src/main/java/io/quarkus/test/junit/virtual/internal/InternalEvents.java:21]: Shall we keep the category name? I am not sure if it is relevant.

**Review by ozangunalp — APPROVED** [review]
It works well!

At first the name of the module confused me as it doesn't do anything related to junit5, but then I saw the lifecycle callbacks were from quarkus-junit5.

We can migrate pinning-test executions under `integration-tests/virtual-threads` to this.

**Comment by cescoffier** [comment]
After discussion, we decided to move the junit5 extension under independent projects, to allow other projects to use it. The functionality is not Quarkus-specific. 

The PR has been updated.

**Review by geoand — APPROVED** [review]
👍🏼

File: test-framework/junit5-virtual-threads/src/main/java/io/quarkus/test/junit/virtual/internal/InternalEvents.java (hunk 1 above)
Comment (cescoffier) [reply test-framework/junit5-virtual-threads/src/main/java/io/quarkus/test/junit/virtual/internal/InternalEvents.java]: Good point, I fixed it!

**Pull request merged**